rotkehlchen/exchanges/okx.py

```diff
@@ -0,0 +1,410 @@
+            location=Location.OKX,
+            api_key=api_key,
+            secret=secret,
+            database=database,
+        )
+        self.passphrase = passphrase
+        self.base_uri = 'https://www.okx.com/'
+        self.msg_aggregator = msg_aggregator
+        self.session.headers.update({
+            'OK-ACCESS-KEY': self.api_key,
+            'OK-ACCESS-PASSPHRASE': self.passphrase,
+        })
+
+    def edit_exchange_credentials(
+            self,
+            api_key: Optional[ApiKey],
+            api_secret: Optional[ApiSecret],
+            passphrase: Optional[str],
+    ) -> bool:
+        changed = super().edit_exchange_credentials(api_key, api_secret, passphrase)
+        if api_key is not None:
+            self.session.headers.update({'OK-ACCESS-KEY': self.api_key})
+        if passphrase is not None:
+            self.session.headers.update({'OK-ACCESS-PASSPHRASE': self.passphrase})
+        return changed
+
+    def _generate_signature(self, timestamp: str, method: str, path: str, body: str) -> str:
+        prehash = timestamp + method.upper() + path + body
+        signature = hmac.new(self.secret, prehash.encode('utf-8'), hashlib.sha256)
+        return base64.b64encode(signature.digest()).decode('utf-8')
+
+    def _api_query(
+        self,
+        endpoint: Literal['currencies', 'balance', 'trades', 'deposits', 'withdrawals'],
+        options: Optional[dict] = None,
+    ) -> dict:
+        method = 'get'
+        options = options.copy() if options else {}
+
+        params = {}
+        if endpoint == 'currencies':
+            path = '/api/v5/asset/currencies'
+        elif endpoint == 'balance':
+            path = '/api/v5/account/balance'
+        elif endpoint == 'trades':
+            path = '/api/v5/trade/orders-history-archive'
+            # supports pagination
+            params.update({
+                'limit': options.get('limit', ''),
+                'after': options.get('after', ''),
+            })
+            params.update({
+                'instType': 'SPOT',
+                'state': 'filled',
+            })
+            if options.get('start_ts'):
+                params['begin'] = str(options['start_ts']) + "000"
+            if options.get('end_ts'):
+                params['end'] = str(options['end_ts']) + "000"
+        elif endpoint == 'deposits':
+            path = '/api/v5/asset/deposit-history'
+            # supports pagination
+            params.update({
+                'limit': options.get('limit', ''),
+                'after': options.get('after', ''),
+            })
+        elif endpoint == 'withdrawals':
+            path = '/api/v5/asset/withdrawal-history'
+            # supports pagination
+            params.update({
+                'limit': options.get('limit', ''),
+                'after': options.get('after', ''),
+            })
+        else:
+            raise AssertionError(f'Unexpected {self.name} endpoint type: {endpoint}')
```

Comment:
Since we use types here and the endpoint param is never set via variables I believe you can remove this error. Also is an exception that you are not handling in other places of the code

Reply:
I understand that static analysis may be able to highlight this error from the type annotation, but python will still accept this as valid code, and throw a different error after a request is made to an invalid endpoint.

How do you suggest I handle the `else` case? I see the following options:

1. Handle one of the listed endpoints in the `else` branch
2. Remove the `else` branch, and initialize `path` to an empty string

Alternatively, if we do keep this, the exception is currently thrown to the caller (in tests) and shown to the end user in the app.